### PR TITLE
SQL pagination should work with the `pretty` parameter

### PR DESCRIPTION
**Pull request by LantaoJin**
### Description
1. Refactoring `isSupported()` improves readability by clearly defining each condition and simplifying the overall logic.
2. `PRETTY` (plus `FORMAT`) parameters should be treated as supported in validating `Cursor` request.
 
### Issues Resolved
resolves https://github.com/opensearch-project/sql/issues/2460
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by penghuo — DISMISSED** [review]

**Review by dai-chen — APPROVED** [review]
Thanks for the fix!

File: sql/src/main/java/org/opensearch/sql/sql/domain/SQLQueryRequest.java [hunk 1]
```diff
@@ -79,19 +82,21 @@ public SQLQueryRequest(
...
-                && noUnsupportedParams
-                && noContent) // if cursor is given, but other things
-            || (noCursor && !noQuery)) // or if cursor is not given, but query
-        && isOnlySupportedFieldInPayload() // and request has supported fields only
-        && isSupportedFormat(); // and request is in supported format
+    boolean hasCursor = isCursor();
+    boolean hasQuery = query != null;
+    boolean hasContent = jsonContent != null && !jsonContent.isEmpty();
+
+    Predicate<String> supportedParams = List.of(QUERY_PARAMS_FORMAT, QUERY_PARAMS_PRETTY)::contains;
```
Comment (vamsi-amazon) [inline sql/src/main/java/org/opensearch/sql/sql/domain/SQLQueryRequest.java:89]: [NIT]won't Set.of would be more appropriate here?

**Comment by vamsi-amazon** [comment]
Any ITs with pretty parameter? Is pretty parameter working?

**Comment by LantaoJin** [comment]
> Any ITs with pretty parameter? Is pretty parameter working?

Currently we don't have a dedicated IT for testing pretty parameter. But we can enabled pretty parameter with this code changes in another PR. https://github.com/opensearch-project/sql/pull/2727/files#diff-8b452f4577c95dae5bcf500b2856b04ace749674765185e81b84daa4ab959098R245

**Comment by LantaoJin** [comment]
Hi @vamsi-amazon @penghuo @dai-chen, could you review and approve this PR again since I addressed comments.

**Comment by LantaoJin** [comment]
None of build issues are related to this code changes. But anyone knows why our CI has lots of failures in recent one or two weeks?

**Review by penghuo — APPROVED** [review]

**Pull request merged**